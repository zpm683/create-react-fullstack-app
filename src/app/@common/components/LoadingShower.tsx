import React, { useState, useEffect } from "react";
import CircularProgress from "@material-ui/core/CircularProgress";
import Snackbar from "@material-ui/core/Snackbar";
import Backdrop from "@material-ui/core/Backdrop";
import { useOnline } from "../middlewares/hooks";
import { selectLoading } from "../redux";
import { useSelector } from "react-redux";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    backdrop: {
      zIndex: theme.zIndex.drawer + 1,
      color: "#fff",
    },
  }),
);

/**
 * Show Snackbar & Backdrop
 *
 * @see @common/middlewares/withLoading.ts
 */
export const LoadingShower = () => {
  const classes = useStyles();
  const isOnline = useOnline();
  const { isLoading, errMsg } = useSelector(selectLoading);
  const [openSnackbar, setOpenSnackbar] = useState(false);
  const [snackbarMsg, setSnackbarMsg] = useState("");

  useEffect(() => {
    console.log(`isOnline:${isOnline}`);
    if (!isOnline) {
      setSnackbarMsg("offline!");
      setOpenSnackbar(true);
    }
  }, [isOnline]);

  useEffect(() => {
    if (errMsg !== "") {
      setSnackbarMsg(`エーラ:${errMsg}`);
      setOpenSnackbar(true);
    }
  }, [errMsg]);
  return (
    <>
      <Snackbar
        data-testid="LoadingShower-Snackbar"
        anchorOrigin={{ vertical: "bottom", horizontal: "right" }}
        open={openSnackbar}
        autoHideDuration={3000}
        onClose={() => setOpenSnackbar(false)}
        message={snackbarMsg}
      />
      <Backdrop
        data-testid="LoadingShower-Backdrop"
        className={classes.backdrop}
        open={isLoading}
      >
        <CircularProgress color="inherit" />
      </Backdrop>
    </>
  );
};
