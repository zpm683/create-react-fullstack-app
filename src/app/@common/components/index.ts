export { LoadingShower } from "./LoadingShower";
export { ErrorBoundary } from "./ErrorBoundary";
