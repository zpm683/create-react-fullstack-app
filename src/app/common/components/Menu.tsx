import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import Tooltip from "@material-ui/core/Tooltip";
import Collapse from "@material-ui/core/Collapse";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ExpandMore from "@material-ui/icons/ExpandMore";
import ChevronRight from "@material-ui/icons/ChevronRight";
import ListItemText from "@material-ui/core/ListItemText";
import { IRootMenu, IMenuInfo } from "common.memu";
import { useHistory } from "react-router-dom";
import Divider from "@material-ui/core/Divider";

const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
  },
  nested: {
    paddingLeft: theme.spacing(4),
  },
}));

interface IOwnProps {
  menuList: IRootMenu[];
}

type MenuProps = IOwnProps;
/**
 * 菜单
 */
export const Menu: React.FC<MenuProps> = ({ menuList }) => {
  const history = useHistory();
  const classes = useStyles();
  const [listState, setListState] = React.useState(true);

  function onClickListItem() {
    const nextListState = !listState;
    setListState(nextListState);
  }

  function renderList(menuList: IMenuInfo[]) {
    return menuList.map((menu, i) => (
      <Tooltip key={i} title={menu.tip}>
        <ListItem
          button
          key={i}
          onClick={() => {
            history.push(`${menu.path}`);
          }}
          className={classes.nested}
        >
          <ListItemText primary={menu.text} />
        </ListItem>
      </Tooltip>
    ));
  }

  return (
    <React.Fragment>
      {menuList.map((menu, i) => {
        return (
          <React.Fragment key={i}>
            <ListItem button onClick={onClickListItem}>
              <ListItemIcon>{menu.icon}</ListItemIcon>
              <ListItemText primary={menu.title} />
              {listState ? <ExpandMore /> : <ChevronRight />}
            </ListItem>
            <Collapse in={listState} timeout="auto" unmountOnExit>
              <List>{renderList(menu.list)}</List>
            </Collapse>
            <Divider />
          </React.Fragment>
        );
      })}
    </React.Fragment>
  );
};
