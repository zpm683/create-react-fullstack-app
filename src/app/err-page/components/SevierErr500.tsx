/**
 * 505 SeverErr
 */

import React from "react";
import { Theme, createStyles, makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";
import { useTranslation } from "react-i18next";
import { ERR_PAGE_KEYS as KEYS } from "../../common/i18n";
import { useGoRootPage } from "../middlewares/customHooks";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: theme.spacing(3, 2),
      width: "50%",
      margin: "0 auto",
    },
  }),
);
export const SevierErr500 = () => {
  const classes = useStyles();
  const { t } = useTranslation();
  const goRootPage = useGoRootPage();

  return (
    <Paper className={classes.root}>
      <Typography variant="h5" component="h3"></Typography>
      <br />
      <Link
        component="p"
        onClick={() => {
          goRootPage();
        }}
      >
        {t(KEYS.GO_TO_ROOT)}
      </Link>
    </Paper>
  );
};
