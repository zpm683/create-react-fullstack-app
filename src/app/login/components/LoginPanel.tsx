import React, { useState } from "react";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Link from "@material-ui/core/Link";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import IMG_LOGO from "../assets/img/logo.png";
import { useHistory } from "react-router";
import { useDispatch, useSelector } from "react-redux";
import { selectToken, selectLoginInfo, doLoginApi } from "../redux/loginSlice";
import { AppPaths } from "../../common";
import { useTranslation } from "react-i18next";
import { LOGIN_KEYS as KEYS } from "../../common";
import { theNextPage } from "../config/loginConfig";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    card: {
      // maxWidth: 345,
    },
    media: {
      width: "192px",
      height: "192px",
      margin: "auto",
      marginTop: "10px",
    },
    textField: {
      width: "100%",
    },
  }),
);

export default function LoginPanel() {
  const classes = useStyles();
  const { t } = useTranslation();
  const loginInfo = useSelector(selectLoginInfo);
  const token = useSelector(selectToken);
  const [userId, changeUserId] = useState(loginInfo.userId);
  const [password, changePassword] = useState(loginInfo.password);
  const history = useHistory();
  const dispatch = useDispatch();

  React.useEffect(() => {
    if (token !== "") {
      history.push(theNextPage);
    }
    // tslint:disable-next-line:align
  }, [token, history]);

  function inputChecker() {
    if (userId.length !== 0 && password.length !== 0) {
      return true;
    } else {
      return false;
    }
  }

  function doLogin() {
    if (inputChecker()) {
      dispatch(doLoginApi({ user: userId, pass: password }));
      // doLoginAction({ userId, password });
    } else {
      alert("入力フォーマットが不正です。");
    }
  }

  function doEnter(keyCode: number) {
    if (keyCode !== 13) return;
    doLogin();
  }

  return (
    <>
      <Card className={classes.card}>
        <CardMedia
          component="img"
          alt="login"
          image={IMG_LOGO}
          title="login"
          className={classes.media}
        />
        <CardContent>
          <TextField
            className={classes.textField}
            label={t(KEYS.USER_NAME)}
            variant="outlined"
            value={userId}
            onChange={e => changeUserId(e.target.value)}
            onKeyPress={e => doEnter(e.which)}
          />
          <br />
          <br />
          <TextField
            type="password"
            className={classes.textField}
            label={t(KEYS.PASSWORD)}
            variant="outlined"
            value={password}
            onChange={e => changePassword(e.target.value)}
            onKeyPress={e => doEnter(e.which)}
          />
        </CardContent>
        <CardContent>
          <Link
            component="button"
            variant="body2"
            onClick={() => {
              alert("Click-Test");
            }}
          >
            {t(KEYS.FORGET_PASSWORD)}
          </Link>
          <br />
          <br />
          <Link
            component="button"
            variant="body2"
            onClick={() => {
              alert("Click-Test");
            }}
          >
            {t(KEYS.REGIST)}
          </Link>
        </CardContent>
        <CardActions title={t(KEYS.LOGIN)}>
          <Button
            fullWidth
            size="large"
            variant="contained"
            color="primary"
            onClick={doLogin}
            onKeyPress={e => doEnter(e.which)}
          >
            {t(KEYS.LOGIN)}
          </Button>
        </CardActions>
      </Card>
    </>
  );
}
