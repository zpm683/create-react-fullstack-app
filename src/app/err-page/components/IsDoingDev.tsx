/**
 * IsDoingDev
 */

import React from "react";
import { Theme, createStyles, makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";
import { ERR_PAGE_KEYS } from "../../@common/i18n";
import { useTranslation } from "react-i18next";
import { useNextPage } from "../../@common/middlewares";
import { cleanToken } from "../../login";
import { useDispatch } from "react-redux";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: theme.spacing(3, 2),
      width: "50%",
      margin: "0 auto",
    },
  }),
);

export const IsDoingDev = () => {
  const classes = useStyles();
  const { t } = useTranslation();
  const nextPage = useNextPage();
  const dispatch = useDispatch();

  return (
    <Paper className={classes.root}>
      <Typography variant="h5" component="h3">
        {t(ERR_PAGE_KEYS.IS_DEVING)}
      </Typography>
      <br />
      <Link
        component="p"
        onClick={() => {
          dispatch(cleanToken());
          nextPage("ROOT");
        }}
      >
        {t(ERR_PAGE_KEYS.GO_TO_ROOT)}
      </Link>
    </Paper>
  );
};
