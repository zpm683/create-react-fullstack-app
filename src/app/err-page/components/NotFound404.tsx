/**
 * 404NotFound
 */

import React from "react";
import { Theme, createStyles, makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";
import { ERR_PAGE_KEYS as KEYS } from "../../@common/i18n";
import { useTranslation } from "react-i18next";
import { useNextPage } from "../../@common/middlewares";
import { cleanToken } from "../../login";
import { useDispatch } from "react-redux";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: theme.spacing(3, 2),
      width: "50%",
      margin: "0 auto",
    },
  }),
);
export const NotFound404 = () => {
  const classes = useStyles();
  const nextPage = useNextPage();
  const dispatch = useDispatch();
  const { t } = useTranslation();

  return (
    <Paper className={classes.root}>
      <Typography variant="h5" component="h3">
        {t(KEYS.NOT_FOUND_404)}
      </Typography>
      <br />
      <Link
        component="p"
        onClick={() => {
          dispatch(cleanToken());
          nextPage("ROOT");
        }}
      >
        {t(KEYS.GO_TO_ROOT)}
      </Link>
    </Paper>
  );
};
