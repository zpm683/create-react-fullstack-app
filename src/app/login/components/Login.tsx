import React from "react";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import { LoginPanel } from "./LoginPanel";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";
import { useTranslation } from "react-i18next";
import { LOGIN_KEYS as KEYS } from "../../common/i18n";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    title: {
      textAlign: "center",
      color: "#ffffff",
    },
    loginRoot: {
      minHeight: "100%",
      width: "100%",
      height: "auto",
      position: "absolute",
      backgroundColor: "#282c34",
    },
    titleContainer: {
      marginTop: 120,
    },
    panelContainer: {
      flexGrow: 1,
      marginTop: 80,
    },
    informationItem: {},
    loginPanelItem: {},
    footerContainer: {
      position: "fixed",
      width: "100%",
      bottom: 0,
    },
    footer: {
      marginTop: "20px",
      textAlign: "center",
      color: "#ffffff",
    },
  }),
);

export const Login = () => {
  const classes = useStyles();
  const { t } = useTranslation();
  return (
    <Box className={classes.loginRoot}>
      <Grid container className={classes.titleContainer}>
        <Grid item xs={12}>
          <Typography className={classes.title} variant="h3">
            {t(KEYS.HELLO)}
          </Typography>
        </Grid>
      </Grid>
      <Grid
        container
        direction="column"
        justify="space-evenly"
        alignItems="center"
        className={classes.panelContainer}
      >
        <Grid item className={classes.loginPanelItem}>
          <LoginPanel />
        </Grid>
      </Grid>
      <Grid item>
        <Typography className={classes.footer} variant="subtitle2">
          DEMO Powered by{" "}
          <Link
            // color="inherit"
            variant="subtitle1"
            href="https://github.com/zpm683/create-react-fullstack-app"
            target="_blank"
          >
            create-react-fullstack-app
          </Link>
        </Typography>
      </Grid>
    </Box>
  );
};
