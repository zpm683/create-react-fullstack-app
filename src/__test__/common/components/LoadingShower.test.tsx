/**
 * app/index.tsx  unit test
 *
 */
import React from "react";
import * as redux from "react-redux";
import { render, fireEvent, act } from "@testing-library/react";
import { LoadingShower } from "../../../app/common/components/LoadingShower";

describe("LoadingShower.tsx unit test.", () => {
  it("Check ChildComponents", () => {
    const useSelectorSpy = jest.spyOn(redux, "useSelector");
    useSelectorSpy.mockReturnValue({ isLoading: true, errMsg: "Err" });
    const { getByTestId } = render(<LoadingShower />);
    expect(getByTestId("LoadingShower-Snackbar")).toBeInTheDocument();
    expect(getByTestId("LoadingShower-Backdrop")).toBeInTheDocument();
    useSelectorSpy.mockReset();
    useSelectorSpy.mockRestore();
  });
});
