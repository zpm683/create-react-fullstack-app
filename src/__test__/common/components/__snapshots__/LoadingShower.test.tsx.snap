// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingShower.tsx unit test. when has no err 1`] = `
<div>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root makeStyles-backdrop-2"
    data-testid="LoadingShower-Backdrop"
    style="opacity: 0; visibility: hidden;"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;
